pkg/collector/corechecks/gpu/gpu.go

```diff
@@ -0,0 +1,203 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package gpu
+
+import (
+	"fmt"
+	"time"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/gpu/model"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	processnet "github.com/DataDog/datadog-agent/pkg/process/net"
+	sectime "github.com/DataDog/datadog-agent/pkg/security/resolvers/time"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+// Check doesn't need additional fields
+type Check struct {
+	core.CheckBase
+	config         *CheckConfig
+	sysProbeUtil   *processnet.RemoteSysProbeUtil
+	lastCheckTime  time.Time
+	timeResolver   *sectime.Resolver
+	statProcessors map[uint32]*StatsProcessor
+}
+
+// Factory creates a new check factory
+func Factory() optional.Option[func() check.Check] {
+	return optional.NewOption(newCheck)
+}
+
+func newCheck() check.Check {
+	return &Check{
+		CheckBase:      core.NewCheckBase(CheckName),
+		config:         &CheckConfig{},
+		statProcessors: make(map[uint32]*StatsProcessor),
+	}
+}
+
+// Parse parses the check configuration
+func (m *CheckConfig) Parse(data []byte) error {
+	return yaml.Unmarshal(data, m)
+}
+
+// Cancel cancels the check
+func (m *Check) Cancel() {
+	ret := nvml.Shutdown()
+	if ret != nvml.SUCCESS {
+		log.Warnf("Failed to shutdown NVML: %v", nvml.ErrorString(ret))
+	}
+}
+
+// Configure parses the check configuration and init the check
+func (m *Check) Configure(senderManager sender.SenderManager, _ uint64, config, initConfig integration.Data, source string) error {
+	if err := m.CommonConfigure(senderManager, initConfig, config, source); err != nil {
+		return err
+	}
+	if err := m.config.Parse(config); err != nil {
+		return fmt.Errorf("ebpf check config: %s", err)
+	}
+	if err := processnet.CheckPath(pkgconfigsetup.SystemProbe().GetString("system_probe_config.sysprobe_socket")); err != nil {
+		return fmt.Errorf("sysprobe socket: %s", err)
```

Comment:
Is this condition expected to be permanent? If not, for example system probe can be expected to start / restart after the core agent, perhaps this check can be moved to `Run`?

Reply:
It's in fact unnecessary as that check is already done in `processnet.GetRemoteSystemProbeUtil`, which is called from `Run`. Removing it, thx for the catch.